examples/ex16.cpp

```diff
@@ -331,8 +353,17 @@ void ConductionOperator::Mult(const Vector &u, Vector &du_dt) const
    // Compute:
    //    du_dt = M^{-1}*-Ku
    // for du_dt, where K is linearized by using u from the previous timestep
-   Kmat.Mult(u, z);
+   //
+   // Essential boundary condition is constant, therefore its time derivative
+   // is set to zero. The system to solve involves only the mass matrix M^{-1},
+   // in consequence we use use this bilinear form to eliminate the boundary
+   // values. The full bilinear form K must be used to diffuse values on the
+   // boundary. If we use Kmat.Mult() only, the interior DoF are disconnected
+   // from the boundary.
+   K->FullMult(u, z);
    z.Neg(); // z = -z
+   du_dt.SetSubVector(ess_tdof_list, 0.0);
+   M->EliminateVDofsInRHS(ess_tdof_list, du_dt, z);
    M_solver.Mult(z, du_dt);
```

Comment:
It makes more sense to me to eliminate boundary conditions in `M` directly, then set `du_dt(tdofs)` to zero here.